server/src/main/java/org/opensearch/action/pagination/WlmPaginationStrategy.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+
+package org.opensearch.action.pagination;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.action.admin.cluster.wlm.WlmStatsResponse;
+import org.opensearch.wlm.stats.WlmStats;
+import org.opensearch.wlm.stats.WorkloadGroupStats;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+
+/**
+ * Pagination strategy for Workload Management (WLM) Stats.
+ * Paginates based on query group IDs.
+ */
+public class WlmPaginationStrategy implements PaginationStrategy<WlmStats> {
+    private static final String DEFAULT_PAGINATED_ENTITY = "wlm_stats";
+
+    private final int pageSize;
+    private final String nextToken;
+    private final SortBy sortBy;
+    private final SortOrder sortOrder;
+    private final List<WlmStats> paginatedStats;
+    private final int snapshotQueryGroupCount;
+    private PageToken responseToken;
+
+    public WlmPaginationStrategy(int pageSize, String nextToken, SortBy sortBy, SortOrder sortOrder, WlmStatsResponse response) {
+        this.pageSize = pageSize;
+        this.nextToken = nextToken;
+        this.sortBy = sortBy;
+        this.sortOrder = sortOrder;
+
+        this.snapshotQueryGroupCount = response.getNodes().stream()
+            .mapToInt(stat -> stat.getWorkloadGroupStats().getStats().size())
+            .sum();
+
+        String currentHash = computeQueryGroupHash(response.getNodes());
```

Comment:
Do we need to calculate this hash every time ? can we add some context around why the has is needed